lib/relay.js

```diff
@@ -43,6 +72,29 @@ function requestHandler(filterRules) {
 
       req.url = result.url;
       logContext.ioUrl = result.url;
+
+      // check if this is a streaming request for binary data
+      if (req.headers && binaryRequest.indexOf(req.headers['accept']) !== -1) {
+        const streamingID = uuid();
+        const streamBuffer = new stream.PassThrough();
+
+        logContext.streamingID = streamingID;
+        logger.debug(logContext, 'sending binary request over websocket connection');
+
+        streams.set(streamingID, streamBuffer);
+        streamBuffer.pipe(res);
+
+        res.locals.io.send('request', {
+          url: req.url,
+          method: req.method,
+          body: req.body,
+          headers: req.headers,
+          streamingID,
+        });
+
+        return;
```

Comment:
We don't set header or anything else in the response as the whole response is streamed directly from the remote system. All filtering must have happened at this point.